p2p/src/net/mod.rs

```diff
@@ -196,3 +193,9 @@ pub trait AsBannableAddress {
 pub trait IsBannableAddress {
     fn is_bannable(&self) -> bool;
 }
+
+#[derive(Debug, Eq, PartialEq)]
+pub enum DisconnectId<A, P> {
```

Comment:
I think this should be done with only peer id... using an address is confusing and is error prone.

Reply:
OK, removed. I still think that disconnecting by socket address is nice to have. Bitcoin Core has it.

Reply:
It's OK to have that in RPC. Then it'll do a reverse-search in peers and disconnect by id. You can even disconnect a subnet. But it shouldn't be the internal implementation by string.